src/libraries/System.Reflection/tests/TypeInfoTests.cs

```diff
@@ -1173,6 +1173,50 @@ public void MakePointerType(Type type)
             Assert.True(pointerType.IsPointer);
         }
 
+        public static IEnumerable<object[]> TypeDescTypesIsPublic_TestData()
+        {
+            yield return new object[] { typeof(int).MakeByRefType(), true };
+            yield return new object[] { typeof(int).MakePointerType(), true };
+            yield return new object[] { typeof(int).MakeArrayType(), true };
+            yield return new object[] { typeof(TI_BaseClass.InternalNestedClass).MakeByRefType(), false };
+            yield return new object[] { typeof(TI_BaseClass.InternalNestedClass).MakePointerType(), false };
+            yield return new object[] { typeof(TI_BaseClass.InternalNestedClass).MakeArrayType(), true };
+            yield return new object[] { typeof(TI_BaseClass.InternalNestedGenericClass<>).MakeGenericType(typeof(string)), false };
+            yield return new object[] { typeof(TI_BaseClass).MakeByRefType(), true };
+            yield return new object[] { typeof(TI_BaseClass).MakePointerType(), true };
+            yield return new object[] { typeof(TI_BaseClass).MakeArrayType(), true };
+        }
+
+        [Theory]
+        [MemberData(nameof(TypeDescTypesIsPublic_TestData))]
+        public void TypeDescTypesIsPublic(Type type, bool expected)
+        {
+            Assert.Equal(expected, type.IsPublic);
+            Assert.Equal(!expected, type.IsNestedAssembly);
+        }
+
+        public delegate void PublicDelegate(string str);
+        internal delegate void InternalDelegate(string str);
+
+        [Fact]
+        public void DelegateTypesIsPublic()
```

Comment:
Please split this into separate `Delegate` and `delegate*` (fcn pointer) tests for clarity since internally they are completely different.

Also adding ```Assert.False(typeof(InternalDelegate).IsPublic)``` would be good.